### PR TITLE
Add ThunderFX as a benchmark executor

### PR DESCRIPTION
**Pull request by riccardofelluga**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

This PR adds ThunderFX in the list of executors in the benchmark suite.

### CODE REVIEW

File: thunder/benchmarks/targets.py [hunk 1]
```diff
@@ -155,11 +156,13 @@ def interpreter_fwd(module: Callable):
     torch_executor,
     torch_compile_executor,
     thunder_executor,
+    thunderfx_executor,
```
Comment (IvanYashchuk) [inline thunder/benchmarks/targets.py:159]: What do we get by using this executor for all benchmarks in this file? All existing benchmarks work fine with plain Thunder.

Comment (riccardofelluga) [reply thunder/benchmarks/targets.py:159]: My idea is to add more information to the benchmarks, that then if not needed can be filtered out by using the `-k` filters. In particular this is useful for the torchbench benchmarks, but I thought that it is also useful for the other existing benchmarks don't you think so?

Comment (riccardofelluga) [reply thunder/benchmarks/targets.py]: Updated and splitted the executors to save on time on the execution of the benchmarks.

**Review by t-vi — APPROVED** [review]
Thank you @riccardofelluga @IvanYashchuk

File: thunder/benchmarks/targets.py [hunk 2]
```diff
@@ -151,17 +152,19 @@ def interpreter_fwd(module: Callable):
     return fn_
 
 
-executors = (
-    torch_executor,
-    torch_compile_executor,
-    thunder_executor,
-)
+executors = (torch_executor, torch_compile_executor, thunder_executor)
```
Comment (IvanYashchuk) [inline thunder/benchmarks/targets.py:155]: What formatter applied this change?

Comment (riccardofelluga) [reply thunder/benchmarks/targets.py:155]: @IvanYashchuk ruff configured by this https://github.com/Lightning-AI/lightning-thunder/blob/40904c762f6b1c2eb1a0c3a75a7b4b1382108b84/pyproject.toml#L84 . Let me know if in your case this doesn't match expectation.

**Pull request merged**